### ISSUE TITLE
Freeze the token embeddings for finetuning

### ISSUE THREAD

**Issue by santhoshkolloju** [huggingface/transformers#5570]
Hi ,

I have finetuned the T5 model using the community notebooks given . But when i looked into the code under examples/seq2seq finetuning code token embedding weights are frozen.

Can some one throw some light on how does this effect the finetuning process?

Thanks

**Comment by patil-suraj** [comment]
freezing embedding weights will result in faster training and will also consume less GPU memory. It might result in slightly less performance but I guess it ultimately depends on the task, the dataset etc.

**Comment by AzharSultan** [comment]
A tangent question: is it possible to freeze only a subset of the token embedding weights? Background: I am adding some new tokens and I want the model to only fine tune on those.

**Comment by montellasebastien** [comment]
It would be a very interesting feature proposed by @AzharSultan like for ***prompt learning*** purpose. 
PyTorch doesn't allow to only update a part of the embeddings lookup table. A solution would be either masking gradients or creating a new lookup table just for the added tokens. Any plan to add that ?

**Comment by lambdaofgod** [comment]
@montellasebastien did you experiment with these workarounds?

**Comment by dhruvdcoder** [comment]
I am also interested in @AzharSultan's question. What is the recommended for doing this right now?
